### PR TITLE
Add testing script to generate anchors

### PR DESCRIPTION
**Pull request by frederikrothenberger**
This PR adds a script to easily generate anchors for II instances with DUMMY_CAPTCHA enabled. This is useful for testing stable memory changes.

### CODE REVIEW

File: scripts/generate-test-anchors [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+SCRIPTS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd "$SCRIPTS_DIR/.."
+
+declare pubkey=
+declare canister_id=
+declare network=
+declare count=
+
+number_check='^[0-9]+$'
```
Comment (nmattia) [inline scripts/generate-test-anchors:13]: this is used in a single place right? why not inline?

File: scripts/generate-test-anchors [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+function help() {
+    cat << EOF
+
+Generates many test anchors on the specified II instance.
+
+NOTE: This requires dfx, hexdump, sed, running II canister with the II_DUMMY_CAPTCHA flavor and a public key file of the dfx identity.
+The public key file can be obtained using
+
+    dfx identity export IDENTITY | openssl ec -out FILE_NAME -pubout -outform der
```
[the remark this answers is not in the document]
Comment (nmattia) [reply scripts/generate-test-anchors:45]: btw this is just the pubkey right? nothing private?

[the remark this answers is not in the document]
Comment (nmattia) [reply scripts/generate-test-anchors:45]: actually why not call `dfx identity export FOO` from this script?

File: scripts/generate-test-anchors [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+fi
+
+if [ -z "$count" ]
+then
+    echo no count
+    usage
+    exit 1
+fi
+
+declare -a network_arg=( )
```
Comment (nmattia) [inline scripts/generate-test-anchors:120]: btw I don't think all those `declare`s are necessary

File: scripts/generate-test-anchors [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+if [ "$network" ]
+then
+    network_arg+=( "--network" "$network")
+fi
+
+function create_anchor {
+    challenge_key=$(dfx canister "${network_arg[@]}" call "$canister_id" create_challenge --candid src/internet_identity/internet_identity.did -qq | sed -n 's/.*challenge_key[[:space:]]*=[[:space:]]*"\(.*\)".*/\1/p')
```
Comment (nmattia) [inline scripts/generate-test-anchors:128]: out of curiosity: is there nothing like `jq` for candid, whom we could tell "grab the `challenge_key` field?

File: scripts/generate-test-anchors [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+    pubkey_blob=$(hexdump -ve '1/1 "%.2x"' "$pubkey" | sed 's/../\\&/g')
+    dfx canister "${network_arg[@]}" call "$canister_id" \
+        register "(record{pubkey=blob \"$pubkey_blob\";alias=\"dfx test key\";purpose=variant{authentication};key_type=variant{unknown};protection=variant{unprotected};},record{key=\"$challenge_key\";chars=\"a\"})" \
+        --candid src/internet_identity/internet_identity.did \
+        -qq
```
Comment (nmattia) [inline scripts/generate-test-anchors:133]: what's `-qq` for? quiet? might as well write `--quiet` to be clear

File: scripts/generate-test-anchors [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+}
+
+function create_anchor_batch {
+    for ((i=0; i < $1; i++))
+    do
+      create_anchor
```
Comment (nmattia) [inline scripts/generate-test-anchors:139]: is this not too slow?

File: scripts/generate-test-anchors [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+    done
+}
+
+SECONDS=0
```
Comment (nmattia) [inline scripts/generate-test-anchors:143]: oh TIL

**Comment by frederikrothenberger** [comment]
Close because it is not a generally useful script (but rather temporarily used infrastructure for testing the stable memory migration).

**Pull request closed without merging**